freva-client/src/freva_client/query.py

```diff
@@ -733,17 +712,111 @@ def _facet_search(
+    def add_user_data(
+        self, username: str, paths: List[str], facets: Dict[str, str]
+    ) -> None:
+        """Add user data to the databrowser.
+
+        Via this functionality, user would be able to add data to the databrowser.
+        It accepts file paths and metadata facets to categorize and store the user's
+        data.
+
+        Parameters
+        ~~~~~~~~~~
+        username: str
+            The username of user.
+        paths: list[str]
+            A list of paths to the data files that should be uploaded or cataloged.
+        facets: dict[str, str]
+            A dictionary containing metadata facets (key-value pairs) to describe the
+            data.
+
+        Returns
+        ~~~~~~~~
+        None
+            If the operation is successful, no return value is provided.
+
+        Raises
+        ~~~~~~~
+        ValueError
+            If the operation fails to add the user data.
+
+        Example
+        ~~~~~~~
+        .. execute_code::
+
+            from freva_client import authenticate, databrowser
+            token_info = authenticate(username="janedoe")
+            db = databrowser()
+            db.add_user_data(
+                "janedoe",
+                ["."],
+                {"project": "cmip5", "experiment": "something"}
+            )
+        """
+        url = f"{self._cfg.userdata_url}/{username}"
+        token = self._auth.check_authentication(auth_url=self._cfg.auth_url)
+        headers = {"Authorization": f"Bearer {token['access_token']}"}
+        params = {"paths": paths}
+        if "username" in facets:
+            del facets["username"]
+        data = facets
+        result = self._put(url, data=data, headers=headers, params=params)
+
+        if result is None:
+            raise ValueError("Failed to add user data")
+
+    def delete_user_data(self, username: str, search_keys: Dict[str, str]) -> None:
+        """
+        Delete user data from the databrowser.
+
+        Uing this, user would be able to delete the user's data from the databrowser
+        based on the provided search keys.
+
+        Parameters
+        ~~~~~~~~~~
+        username: str
+            The username associated with the data to be deleted.
+        search_keys: dict[str, str]
+            A dictionary containing the search keys to identify the data to be deleted.
+
+        Returns
+        ~~~~~~~~
+        None
+            If the operation is successful, no return value is provided.
+
+        Raises
+        ~~~~~~~
+        ValueError
+            If the operation fails to delete the user data.
+
+        Example
+        ~~~~~~~
+        .. execute_code::
+
+            from freva_client import databrowser, authenticate
+            token_info = authenticate(username="janedoe")
+            db = databrowser()
+            db.delete_user_data(
+                "janedoe",
+                {"project": "cmip5", "experiment": "something"}
+            )
```

Comment:
I haven't look at it yet, but if the restAPI supports wild-cards you can also show it here.